enumerable.rb

```diff
@@ -0,0 +1,106 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    arr = is_a?(Range) ? to_a : self
+    i = 0
+    while i < arr.size
+      yield(arr[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    arr = is_a?(Range) ? to_a : self
+    i = 0
+    while i < arr.size
+      yield(arr[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_count
+    return length unless block_given?
+
+    arr = is_a?(Range) ? to_a : self
+    count = 0
+    i = 0
+    while i < arr.size
+      count += 1 if yield(arr[i])
+      i += 1
+    end
+    count
+  end
+
+  def my_select
+    return enum_for unless block_given?
+
+    arr = is_a?(Range) ? to_a : self
+    count = []
+    i = 0
+    while i < arr.size
+      count << arr[i] if yield(arr[i])
+      i += 1
+    end
+    count
+  end
+
+  def my_all?
+    arr = is_a?(Range) ? to_a : self
+    i = 0
+    while i < arr.size
+      yield(arr[i])
+      return false unless yield(arr[i])
+
+      i += 1
+    end
+    true
+  end
+
+  def my_any?(con = false)
+    if block_given?
+      size.times do |item|
+        result = yield(self[item])
+        (return true) if result
+      end
+    elsif con
+      size.times do |item|
+        return true if con == self[item]
+      end
+    else
+      size.times do |item|
+        return true if self[item]
+      end
+    end
+    false
+  end
+
+  def my_none?
+    unless block_given?
+      my_each { |item| return false if item != false }
+      return true
+    end
+    my_each do |item|
+      return false if yield(item) == true
+    end
+    true
+  end
+
+  def my_map(element = nil)
+    new_array = []
+
+    my_each do |item|
+      new_array << if block_given?
+                     yield(item)
+                   else
+                     element.call(item)
+                   end
+    end
+
+    new_array
+  end
```

Comment:
* [ ] should have working `my_map` identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_map 

- [ ] returns an Enumerator if no block is given
```ruby
[2,4,7,11].my_map == [2,4,7,11].map #<Enumerator: [2, 4, 7, 11]:map
```
- Yours is retrieving an NoMethodError
![image](https://user-images.githubusercontent.com/52209067/79677264-b63c8900-81ac-11ea-92e7-b615db07b944.png)

- Real map
![image](https://user-images.githubusercontent.com/52209067/77697646-dc3b8880-6f74-11ea-9277-4ed4a2b3af4c.png)

- [ ] in the specifications of the project in the [Odin project](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks) # 12, you were asked to modify your my_map method to take either a proc or a block. And if both a proc and a block are given, only execute the proc. The real map doesn't have this feature, but you were asked to have it so let's work on it

Given 
```ruby
my_proc = Proc.new {|num| num > 10 }
```
Result
```ruby
puts [18, 22, 5, 6] .my_map(my_proc) {|num| num < 10 }  => true true false false
```